tasks/deploy/hyperdrive-matching-engine.ts

```diff
@@ -0,0 +1,92 @@
+import { subtask } from "hardhat/config";
+import {
+    Address,
+    decodeEventLog,
+    encodeDeployData,
+    encodePacked,
+    parseEther,
+} from "viem";
+import { CREATEX_ABI } from "../../lib/createx/interface/src/lib/constants";
+import {
+    CREATE_X_FACTORY,
+    CREATE_X_FACTORY_DEPLOYER,
+    CREATE_X_PRESIGNED_TRANSACTION,
+    HYPERDRIVE_MATCHING_ENGINE_SALT,
+    HyperdriveDeployBaseTask,
+    HyperdriveDeployNamedTaskParams,
+} from "./lib";
+
+// Extend params to include the additional constructor arguments
+export interface DeployHyperdriveMatchingEngineParams
+    extends HyperdriveDeployNamedTaskParams {
+    morpho: Address;
+}
+
+HyperdriveDeployBaseTask(
+    subtask(
+        "deploy:hyperdrive-matching-engine",
+        "deploys the HyperdriveMatchingEngine contract to the configured chain",
+    ),
+).setAction(
+    async (
+        { name, morpho }: DeployHyperdriveMatchingEngineParams,
+        { hyperdriveDeploy, artifacts, viem, network },
+    ) => {
+        console.log("\nRunning deploy:hyperdrive-matching-engine ...");
+
+        if (network.name == "anvil") {
+            let tc = await viem.getTestClient();
+            await tc.setBalance({
+                value: parseEther("1"),
+                address: CREATE_X_FACTORY_DEPLOYER,
+            });
+            let wc = await viem.getWalletClient(CREATE_X_FACTORY_DEPLOYER);
+            await wc.sendRawTransaction({
+                serializedTransaction: CREATE_X_PRESIGNED_TRANSACTION,
+            });
+        }
```

Comment:
will this revert if the CreateX factory already exists? wondering if we should have a check here...

also seeing this logic block in a couple places, wdyt of consolidating into a shared function?

Reply:
Good catch!